packages/front-end/components/Experiment/ResultsDownloadButton.tsx

```diff
@@ -0,0 +1,121 @@
+import { ExperimentInterfaceStringDates } from "back-end/types/experiment";
+import { ExperimentReportResultDimension } from "back-end/types/report";
+import React, { useMemo } from "react";
+import { FaFileExport } from "react-icons/fa";
+import { useDefinitions } from "../../services/DefinitionsContext";
+import { ExperimentTableRow, getRisk } from "../../services/experiments";
+
+type UpdatedRow = {
+  metricName?: string;
+};
+
+export default function ResultsDownloadButton({
+  results,
+  experiment,
+}: {
+  results?: ExperimentReportResultDimension;
+  experiment: ExperimentInterfaceStringDates;
+}) {
+  const { getMetricById, ready } = useDefinitions();
+
+  const csvRows = [];
+
+  const rows = useMemo<ExperimentTableRow[]>(() => {
+    if (!results || !results.variations || !ready) return [];
+    return experiment.metrics
+      .map((m) => {
+        const metric = getMetricById(m);
+        return {
+          label: metric?.name,
+          metric,
+          rowClass: metric?.inverse ? "inverse" : "",
+          variations: results.variations.map((v) => {
+            return v.metrics[m];
+          }),
+        };
+      })
+      .filter((row) => row.metric);
+  }, [results, ready]);
+
+  rows.forEach((row) => {
+    row.variations.forEach((variant, index) => {
+      variant.name = experiment.variations[index].name;
+      const { relativeRisk } = getRisk(index, row);
+      variant.relativeRisk = relativeRisk;
+    });
+  });
+
+  rows.forEach((row) => {
+    const updatedRow: UpdatedRow = {};
+    const metric = getMetricById(row.metric.id);
+    updatedRow.metricName = metric.name;
+
+    row.variations.forEach((variation) => {
```

Comment:
This is another piece that I don't love - it's not very dynamic, which I feel like makes it fragile.